homeassistant/components/somfy/cover.py

```diff
@@ -52,15 +62,27 @@ async def async_update(self):
 
     def close_cover(self, **kwargs):
         """Close the cover."""
-        if self.optimistic:
+        self._is_closing = True
+        self.schedule_update_ha_state()
+        try:
+            # Blocks until the close command is sent
+            self.cover.close()
             self._closed = True
-        self.cover.close()
+        finally:
+            self._is_closing = None
+            self.schedule_update_ha_state()
 
     def open_cover(self, **kwargs):
         """Open the cover."""
-        if self.optimistic:
+        self._is_opening = True
+        self.schedule_update_ha_state()
```

Comment:
That’s to force an immediate update of `is_opening`?

Reply:
Yes

Reply:
Realized we should probably use `async_write_ha_state`. Untested suggestion here https://github.com/home-assistant/core/pull/42448